### DISCUSSION TITLE
Batch metadata

### DISCUSSION THREAD

**Discussion by arcanis** [npm/feedback#88] [Registry]
Requesting the metadata for a set of packages currently requires to make one http request for each package. It would be interesting to see whether a dedicated "batched request" endpoint would help making tree resolutions faster (perhaps not, harder caching etc, but I would at least be very interested to read the results of such an investigation).

Ref: https://twitter.com/arcanis/status/1087045722703347712

**Comment by MylesBorins** [comment]
@arcanis are you primarily looking at getting a batch of packuments at once?

**Reply by arcanis** [reply]
Mostly, yep. To give you an idea, installing Gatsby triggers 1523 requests, spread across 13 passes - some of them with a single requests, so it's more like 8-9 effective passes. There's a decent chance that being able to retrieve the information for 150 packages at a time could lead to interesting improvements when generating lockfiles from scratch.
